packages/node_modules/@webex/plugin-meetings/src/networkQualityMonitor/index.js

```diff
@@ -0,0 +1,178 @@
+import EventsScope from '../common/events/events-scope';
+import {EVENT_TRIGGERS, STATS} from '../constants';
+
+
+/**
+  * Meeting - network quality judgment event
+  * Emitted on each interval of retrieving stats Analyzer data
+  * @event network:quality:judgment
+  * @type {Object}
+  * @property {string} mediaType {video|audio}
+  * @property {boolean} networkAcceptable is network overal acceptable
+  * @property {Object} networkQualityDataStructure for each indicator type
+  * @property {boolean} networkQualityDataStructure.acceptable
+  * @property {Number} networkQualityDataStructure.value - score or RTC value for each indicator type
+  * @memberof NetworkQualityMonitor
+  */
+/**
+ * NetworkQualityMonitor class that will emit events based on detected quality
+ *
+ * @class NetworkQualityMonitor
+ * @extends {EventsScope}
+ */
+export default class NetworkQualityMonitor extends EventsScope {
+  /**
+   * Creates a new instance of NetworkQualityMonitor
+   * @constructor
+   * @public
+   * @param {Object} config
+   */
+  constructor(config) {
+    super();
+    this.config = config;
+    this.indicatorTypes = Object.freeze({
+      PACKETLOSS: 'packetLoss',
+      LATENCY: 'latency',
+      JITTER: 'jitter'
+    });
+    this.adequeteExistingNetworkQuality = true;
+    this.networkQualityDataStructure = {};
+    this.mediaType = null;
+  }
+
+  /**
+ * emits NETWORK_QUALITY_JUDGMENT event on meeting with payload
+ *
+ * @memberof NetworkQualityMonitor
+ * @returns {void}
+ */
+  emitNetworkJudgment() {
+    this.emit(
+      {
+        file: 'networkQualityMonitor',
+        function: 'emitNetworkJudgment'
+      },
+      EVENT_TRIGGERS.NETWORK_QUALITY_JUDGMENT,
+      {
+        mediaType: this.mediaType,
+        networkAcceptable: this.adequeteExistingNetworkQuality,
```

Comment:
does this indicate if the values is below threshold ? Can we just do 1 or 0 ?

Reply:
I had it as a boolean - if we use 1 or 0 I thought that might be confusing - for example if jitter has a zero value - that's good in our case and networkAcceptable is true but if jitter is 0 and networkAcceptable is 1 it might be more confusing? also in JS 0 == false is truthy
but 0 === false is falsy
so I though just using a boolean would be easier for the user.